plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -237,35 +239,56 @@ public static ReaderPageSource createPageSource(
+            Optional<FilterPredicate> filterPredicate;
+            if (!options.isIgnoreStatistics()) {
+                domainAndPredicates = disjunctTupleDomains.stream()
+                        .map(domain -> {
+                            TupleDomain<ColumnDescriptor> parquetTupleDomain = getParquetTupleDomain(descriptorsByPath, domain, fileSchema, useColumnNames);
+                            TupleDomainParquetPredicate parquetPredicate = buildPredicate(requestedSchema, parquetTupleDomain, descriptorsByPath, timeZone);
+                            return new ParquetDomainAndPredicate(parquetTupleDomain, parquetPredicate);
+                        })
+                        .collect(toImmutableList());
+
+                filterPredicate = domainAndPredicates.stream()
+                        .map(ParquetDomainAndPredicate::predicate)
+                        .map(parquetPredicate -> parquetPredicate.toParquetFilter(timeZone))
+                        .flatMap(Optional::stream)
+                        // TODO create balanced tree when this is used with high number of filters.
+                        .reduce(FilterApi::or);
+            }
+            else {
+                domainAndPredicates = ImmutableList.of();
+                filterPredicate = Optional.empty();
+            }
 
             long nextStart = 0;
             ImmutableList.Builder<BlockMetaData> blocks = ImmutableList.builder();
             ImmutableList.Builder<Long> blockStarts = ImmutableList.builder();
             ImmutableList.Builder<Optional<ColumnIndexStore>> columnIndexes = ImmutableList.builder();
             for (BlockMetaData block : parquetMetadata.getBlocks()) {
                 long firstDataPage = block.getColumns().get(0).getFirstDataPageOffset();
-                Optional<ColumnIndexStore> columnIndex = getColumnIndexStore(dataSource, block, descriptorsByPath, parquetTupleDomain, options);
-                Optional<BloomFilterStore> bloomFilterStore = getBloomFilterStore(dataSource, block, parquetTupleDomain, options);
-
-                if (start <= firstDataPage && firstDataPage < start + length
-                        && predicateMatches(
-                        parquetPredicate,
-                        block,
-                        dataSource,
-                        descriptorsByPath,
-                        parquetTupleDomain,
-                        columnIndex,
-                        bloomFilterStore,
-                        timeZone,
-                        domainCompactionThreshold)) {
-                    blocks.add(block);
-                    blockStarts.add(nextStart);
-                    columnIndexes.add(columnIndex);
+                for (ParquetDomainAndPredicate domain : domainAndPredicates) {
+                    TupleDomain<ColumnDescriptor> parquetTupleDomain = domain.tupleDomain;
+                    TupleDomainParquetPredicate parquetPredicate = domain.predicate;
```

Comment:
Just use the record accessors directly wherever needed instead of extracting a variable